metatensor-torch/src/labels.cpp

```diff
@@ -218,11 +176,43 @@ TorchLabels LabelsHolder::view(const TorchLabels& labels, std::vector<std::strin
     return torch::make_intrusive<LabelsHolder>(std::move(names), std::move(new_values), CreateView{});
 }
 
-LabelsHolder::LabelsHolder(metatensor::Labels labels):
-    names_(names_from_metatensor(labels)),
-    values_(values_from_metatensor(labels)),
-    labels_(std::move(labels))
-{}
+LabelsHolder::LabelsHolder(metatensor::Labels labels): labels_(std::move(labels)) {
+    // extract the names
+    for (const auto* name: this->labels_->names()) {
+        this->names_.emplace_back(name);
+    }
+
+    // check if the labels are already associated with a tensor
+    auto* user_data = this->labels_->user_data();
+    if (user_data != nullptr) {
+        // if we start using user_data for more than this exact case (storing
+        // tensors inside Labels), this code might fails and will need to start
+        // checking that `user_data` is actually a tensor.
+        this->values_ = *static_cast<torch::Tensor*>(user_data);
```

Comment:
It is not really part of this PR, but since I already spent time to review it (I think this is even the second time), maybe the text can be improved?

I don't really understand the last part
```
ànd will need do start checking "user_data" is actually a tensor`
```
Do you mean the `static_cast` below? Why does the `static_cast` help?

Reply:
static_cast does not help, and here we assume that if user_data exists, then it is a torch::Tensor. I've rephrased this comment, let me know if it helps!